packages/wrangler/src/cloudchamber/deploy.ts

```diff
@@ -0,0 +1,168 @@
+import { readFileSync, statSync } from "fs";
+import path from "path";
+import { type Config } from "../config";
+import { type ContainerApp } from "../config/environment";
+import { UserError } from "../errors";
+import { CI } from "../is-ci";
+import { type Logger } from "../logger";
+import { fetchVersion } from "../versions/api";
+import { apply } from "./apply";
+import { build } from "./build";
+import { fillOpenAPIConfiguration } from "./common";
+
+export async function buildContainers(config: Config, workerTag: string) {
+	if (config.containers === undefined) {
+		return;
+	}
+
+	for (const container of config.containers) {
+		const options = getBuildArguments(container, workerTag, true);
+		if (options.isDockerImage) {
+			await build(options);
+		}
+	}
+}
+
+export async function deployContainers(
+	logger: Logger,
+	config: Config,
+	{
+		versionId,
+		accountId,
+		scriptName,
+		dryRun,
+		env,
+	}: {
+		versionId: string;
+		accountId: string;
+		scriptName: string;
+		dryRun?: boolean;
+		env?: string;
+	}
```

Comment:
Why does this function take some arguments as a bag of options, and `config` separately?

Reply:
configuration of the container app and build specific arguments baiscally